Xamarin.Forms.Platform.Android/Platform.cs

```diff
@@ -1238,8 +1208,8 @@ SizeRequest IPlatform.GetNativeSize(VisualElement view, double widthConstraint,
 			IVisualElementRenderer viewRenderer = GetRenderer(view);
 
 			// negative numbers have special meanings to android they don't to us
-			widthConstraint = widthConstraint <= -1 ? double.PositiveInfinity : _context.ToPixels(widthConstraint);
-			heightConstraint = heightConstraint <= -1 ? double.PositiveInfinity : _context.ToPixels(heightConstraint);
+			widthConstraint = widthConstraint <= -1 ? double.PositiveInfinity : _activity.ToPixels(widthConstraint);
```

Comment:
For all those cases where a `Context` instance is sufficient it should be used instead of trying to use an `Activity` instance (i.e. you should store the base `Context` instance and then either also it's potentially upcasted `Activity` counterpart or a property that tries to upcast for you). Otherwise we now traded an `InvalidCastException` for `NullReferenceException` peppered everywhere.